test/contrib/test_pyopenssl_dependencies.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+import unittest
+
+from nose.plugins.skip import SkipTest
+
+try:
+    from urllib3.contrib.pyopenssl import inject_into_urllib3
+except ImportError as e:
+    raise SkipTest('Could not import PyOpenSSL: %r' % e)
+
+from mock import patch
+
+class TestPyOpenSSLInjection(unittest.TestCase):
+    """
+    Tests for error handling in pyopenssl's 'inject_into urllib3'
+    """
+    def test_inject_validate_fail(self):
+        """
+        Injection should not be supported if we are missing required dependencies.
+        """
+        with patch("cryptography.x509.extensions.Extensions") as mock:
+            del mock.get_extension_for_class
+            self.assertRaises(ImportError, inject_into_urllib3)
```

Comment:
Might be the better way to go though. =)

The issue isn't "can we call `inject_into_urllib3` more than once", it's that if we call into it and it succeeds, and we run this test before we run any of the others, all the other tests are now running against PyOpenSSL instead of the stdlib. That's not a good outcome.